backend/services/mgmt/v1alpha1/connection-data-service/connection-data.go

```diff
@@ -388,12 +389,101 @@ func (s *Service) GetConnectionDataStream(
 		if err != nil {
 			return fmt.Errorf("unable to finish sending record stream: %w", err)
 		}
+	case *mgmtv1alpha1.ConnectionConfig_DynamodbConfig:
+		dynamoclient, err := s.awsManager.NewDynamoDbClient(ctx, config.DynamodbConfig)
+		if err != nil {
+			return fmt.Errorf("unable to create dynamodb client from connection: %w", err)
+		}
+		var lastEvaluatedKey map[string]dynamotypes.AttributeValue
+
+		for {
+			output, err := dynamoclient.ScanTable(ctx, req.Msg.Table, lastEvaluatedKey)
+			if err != nil {
+				return fmt.Errorf("failed to scan table %s: %w", req.Msg.Table, err)
+			}
+
+			for _, item := range output.Items {
+				row := make(map[string][]byte)
+
+				itemBits, err := convertMapToJSONBytes(item)
+				if err != nil {
+					return err
+				}
+				row["item"] = itemBits
+				if err := stream.Send(&mgmtv1alpha1.GetConnectionDataStreamResponse{Row: row}); err != nil {
+					return fmt.Errorf("failed to send stream response: %w", err)
+				}
+			}
+
+			lastEvaluatedKey = output.LastEvaluatedKey
+			if lastEvaluatedKey == nil {
+				break
+			}
+		}
+
 	default:
 		return nucleuserrors.NewNotImplemented(fmt.Sprintf("this connection config is not currently supported: %T", config))
 	}
 	return nil
 }
 
+func convertAttributeValueToDynamoDBJSON(av dynamotypes.AttributeValue) (map[string]any, error) {
```

Comment:
Do we have this method somewhere else? Meaning: are we copying this anywhere or is this bespoke to the connection data? If so, we should centralize it...